pkg/controller/instance/instance_controller.go

```diff
@@ -258,18 +258,44 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 			}
 
 			for _, instance := range instances.Items {
-				log.Printf("InstanceController: Queing instance %v for reconciliation", instance.Name)
-				requests = append(requests, reconcile.Request{
-					NamespacedName: types.NamespacedName{
-						Name:      instance.Status.ActivePlan.Name,
-						Namespace: instance.Status.ActivePlan.Namespace,
-					},
-				})
+				// Sanity check - lets make sure that this instance references the frameworkVersion
+				if instance.Spec.FrameworkVersion.Name == a.Meta.GetName() &&
+					instance.Spec.FrameworkVersion.Namespace == a.Meta.GetNamespace() &&
+					instance.Status.ActivePlan.Name == "" {
+
+					log.Printf("InstanceController: Creating a deploy execution plan for the instance %v", instance.Name)
+					err = createPlan(mgr, "deploy", &instance)
+					if err != nil {
+						log.Printf("InstanceController: Error creating \"%v\" object for \"deploy\": %v", instance.Name, err)
+					}
+
+					log.Printf("InstanceController: Queing instance %v for reconciliation", instance.Name)
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name:      instance.Name,
+							Namespace: instance.Namespace,
+						},
+					})
+				}
 			}
+			log.Printf("Found %v instances to reconcile", len(requests))
 			return requests
 		})
 
-	if err = c.Watch(&source.Kind{Type: &kudov1alpha1.FrameworkVersion{}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: mapFn}); err != nil {
+	// This map function makes sure that we *ONLY* handle created frameworkVersion
```

Comment:
This brings up a bigger question that's out of scope for here:

IF someone updates a FV, what do we do with the instances?

Reply:
Lots of implications here. The instance might drastically change on the next plan run, even if we didn't do anything. Maybe we need to serialize the current FV to the instance.Status or something, idk.

Reply:
Yep, the same race is certainly possible with FV+I update.